src/dlr_relayvm.cc

```diff
@@ -0,0 +1,283 @@
+#include "dlr_relayvm.h"
+
+
+#include <stdlib.h>
+#include <fstream>
+#include <iterator>
+#include <numeric>
+
+using namespace dlr;
+
+const std::string RelayVMModel::ENTRY_FUNCTION = "main";
+
+void RelayVMModel::InitModelPath(std::vector<std::string> paths) {
+  path_ = std::make_unique<ModelPath>();
+  std::vector<std::string> paths_vec;
+  for (auto path : paths) {
+    ListDir(path, paths_vec);
+  }
+
+  for(auto path: paths_vec) {
+    if (!EndsWith(path, LIBDLR) && EndsWith(path, ".so")) {
+      path_->model_lib = path;
+    } else if (EndsWith(path, ".ro")) {
+      path_->relay_executable = path;
+    } else if (EndsWith(path, ".meta")) {
+      path_->metadata = path;
+    }
+  }
+
+  if (path_->model_lib.empty() || path_->relay_executable.empty() || path_->metadata.empty()) {
+    LOG(FATAL) << "Invalid model artifact!";
+  }
+}
+
+void RelayVMModel::SetupVMModule() {
+    tvm::runtime::Module lib = tvm::runtime::Module::LoadFromFile(path_->model_lib);
+    std::ifstream relay_ob(path_->relay_executable, std::ios::binary);
+    std::string code_data((std::istreambuf_iterator<char>(relay_ob)), std::istreambuf_iterator<char>());
+
+    vm_executable_ = std::make_shared<tvm::runtime::Module>(tvm::runtime::vm::Executable::Load(code_data, lib));
+    auto vm = tvm::runtime::make_object<tvm::runtime::vm::VirtualMachine>();
+    vm->LoadExecutable(static_cast<tvm::runtime::vm::Executable*>(const_cast<tvm::runtime::Object*>(vm_executable_->get())));
```

Comment:
Shouldn't we be able to cast to Executable* with just one casting

Reply:
no it returns a const pointer

Reply:
vm_executable_->get() returns a constant.